### PR TITLE
Stefan's newsfeed ideas

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds improvements to the newsfeed checking based on Stefan's ideas on the first PR.

### CODE REVIEW

**Comment by stefanv** [comment]
More coming, presumably :)

**Comment by mcoughlin** [comment]
@stefanv yeah i think i just got your last idea.

**Review by stefanv — APPROVED** [review]
Thank you for the refactoring, @mcoughlin!

File: skyportal/handlers/api/news_feed.py [hunk 1]
```diff
@@ -56,24 +56,18 @@ def get(self):
...
-                n_items = min(
-                    int(preferences['newsFeed']['numItems']), DEFAULT_NEWSFEED_ITEMS
-                )
-            else:
-                n_items = DEFAULT_NEWSFEED_ITEMS
-        else:
-            n_items = int(n_items)
-
+            n_items_query = None
+        n_items_feed = preferences.get('newsfeed', {}).get('numItems', None)
```
[the remark this answers is not in the document]
Comment (stefanv) [reply skyportal/handlers/api/news_feed.py]: Ignore; current version looks good.

File: skyportal/handlers/api/news_feed.py [hunk 2]
```diff
@@ -56,24 +56,18 @@ def get(self):
...
-            if 'newsFeed' in preferences and 'numItems' in preferences['newsFeed']:
-                n_items = min(
-                    int(preferences['newsFeed']['numItems']), DEFAULT_NEWSFEED_ITEMS
-                )
-            else:
-                n_items = DEFAULT_NEWSFEED_ITEMS
-        else:
-            n_items = int(n_items)
-
+            n_items_query = None
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/news_feed.py:64]: ah yup.

**Pull request merged**